app/src/main/java/com/lightricks/feedexercise/ui/MainActivity.kt

```diff
@@ -1,15 +1,77 @@
 package com.lightricks.feedexercise.ui
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import com.lightricks.feedexercise.R
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.ViewModelProvider
+import com.lightricks.feedexercise.appComponent
+import com.lightricks.feedexercise.ui.feed.*
+import com.lightricks.feedexercise.ui.feed.model.ViewState
+import javax.inject.Inject
 
 /**
  * This is the main entry point into the app. This activity shows the [FeedFragment].
  */
-class MainActivity : AppCompatActivity() {
+class MainActivity : ComponentActivity() {
+
+    @Inject
+    lateinit var feedViewModelFactory: FeedViewModelFactory
+
+    private val viewModel: FeedViewModel by lazy {
+        ViewModelProvider(this, feedViewModelFactory)[FeedViewModel::class.java]
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
+        appComponent().inject(this)
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.main_activity)
+        setContent {
+            MaterialTheme {
+                FeedScreen(viewModel)
+            }
+        }
+    }
+
+    @Composable
+    fun FeedScreen(viewModel: FeedViewModel) {
+        val viewState = viewModel.viewState.collectAsState()
+        when (val state = viewState.value) {
+            ViewState.Loading -> LoadingState()
+            is ViewState.Error -> ErrorState(state, viewModel)
+            is ViewState.Content -> ContentState(content = state, onSwipeToRefresh = viewModel::onRefresh)
+        }
+    }
+
+    @Preview
+    @Composable
+    fun DefaultPreview() {
+        MaterialTheme {
+//        FeedScreen()
+            Box(modifier = Modifier.fillMaxSize()) {
+                Column(Modifier.align(Alignment.Center)) {
+                    Text(text = "Something went wrong!")
+                    TextButton(
+                        modifier = Modifier
+                            .align(Alignment.CenterHorizontally)
+                            .padding(top = 4.dp),
+                        onClick = { viewModel.onRefresh() }
+                    ) {
+                        Text(text = "Reload")
+                    }
+                }
+            }
+        }
```

Comment:
remove?

Reply:
This is good during debug not to merge in the code base.

Reply:
Its like using tools:xxx inside layout. But doesnt matter - I will remove it

Reply:
IMO this pollute the code, I am not sure how we should work with compose, but for sure I will not want to see a code that is there for debug as part of the code base. 

And no I don't think it is the same, in xml it is very clear what is for debug and what is not, here you add code that looks exactly like a regular compose function that can be use by mistake in production.